### ISSUE TITLE
Support for send array of files in the same value.

### ISSUE THREAD

**Issue by adrian-netapay** [lostisland/faraday#1169]
## Basic Info

* Faraday Version: 1.0.1
* Ruby Version: 2.6.6

## Issue description
I need to send an array of attachments in the same value. But at the time of the request you do not send any file. With one file works fine.

## Steps to reproduce
If possible, please provide the steps to reproduce the issue.

```ruby
def attachment
if @transaction.product_id == ViewTransaction::MINI

  return [Faraday::FilePart.new(File.open(CreatePdfAction.new(@transaction,'tickets').execute),
                                       'application/pdf',
                                       File.basename("Ticket")),
           Faraday::FilePart.new(File.open(CreatePdfAction.new(@transaction,'vouchers').execute),
                                               'application/pdf',
                                               File.basename("Voucher"))]
else
  return Faraday::FilePart.new(File.open(File.open(CreatePdfAction.new(@transaction,'vouchers').execute),
                        'application/pdf',
                        File.basename(template))
end

def payload(payload = {})
  payload[:attachment] = attachment
  payload[:data] = data
  payload
end

conn = Faraday.new() do |f|
  f.request :multipart
  f.adapter :net_http
end
response = conn.post("http://email-service/v1/email", payload)
```
I don't know if it's possible to add support for sending a file array. Thanks

**Comment by iMacTia** [comment]
@adrian-netapay so first of all let's see what is coming out from Faraday by adding the logger middleware.
Please place the following in your connection setup, between the `:multipart` middleware and the adapter:

```
faraday.response :logger, nil, { headers: true, bodies: true }
```

After this, try the request again and post here the logs please so we can have a look 👍

**Comment by adrian-netapay** [comment]
Thanks for the prompt reply.

This is the output:

App 5065 output: I, [2020-07-21T16:33:58.635055 #5065]  INFO -- request: POST http://email-service/v1/email
App 5065 output: I, [2020-07-21T16:33:58.635320 #5065]  INFO -- request: User-Agent: "Faraday v1.0.1"
App 5065 output: Content-Type: "multipart/form-data; boundary=-----------RubyMultipartPost-642b1a8c3f4afb7658ab808cc646bc9e"
App 5065 output: Content-Length: "71387"
App 5065 output: I, [2020-07-21T16:33:58.641619 #5065]  INFO -- request: #<Faraday::CompositeReadIO:0x00007f96d9fd8348
App 5065 output:  @index=0,
App 5065 output:  @ios=
App 5065 output:   [#<CompositeReadIO:0x00007f96d9fd8a78
App 5065 output:     @index=0,
App 5065 output:     @ios=
App 5065 output:      [#<StringIO:0x00007f96d9fd8ac8>,
App 5065 output:       #<UploadIO:0x00007f96d9f32f60
App 5065 output:        @content_type="application/pdf",
App 5065 output:        @io=
App 5065 output:         #<File:/vagrant_ruby/martian-library/pdfs/Ticket-128DBBAE-60D4-8CF2-0D8B-7D9E3A9B912D.pdf>,
App 5065 output:        @local_path=
App 5065 output:         "/vagrant_ruby/martian-library/pdfs/Ticket-128DBBAE-60D4-8CF2-0D8B-7D9E3A9B912D.pdf",
App 5065 output:        @opts={},
App 5065 output:        @original_filename="Ticket">,
App 5065 output:       #<StringIO:0x00007f96d9fd8aa0>]>,
App 5065 output:    #<CompositeReadIO:0x00007f96d9fd8640
App 5065 output:     @index=0,
App 5065 output:     @ios=
App 5065 output:      [#<StringIO:0x00007f96d9fd8690>,
App 5065 output:       #<UploadIO:0x00007f96d9fdb750
App 5065 output:        @content_type="application/pdf",
App 5065 output:        @io=
App 5065 output:         #<File:/vagrant_ruby/martian-library/pdfs/Voucher-128DBBAE-60D4-8CF2-0D8B-7D9E3A9B912D.pdf>,App 5065 output:        @local_path=
App 5065 output:         "/vagrant_ruby/martian-library/pdfs/Voucher-128DBBAE-60D4-8CF2-0D8B-7D9E3A9B912D.pdf",
App 5065 output:        @opts={},
App 5065 output:        @original_filename="Voucher">,
App 5065 output:       #<StringIO:0x00007f96d9fd8668>]>,
App 5065 output:    #<StringIO:0x00007f96d9fd83e8>,
App 5065 output:    #<StringIO:0x00007f96d9fd8370>],
App 5065 output:  @parts=
App 5065 output:   [#<Parts::FilePart:0x00007f96d9fd8d70
App 5065 output:     @foot="\r\n",
App 5065 output:     @head=
App 5065 output:      "-------------RubyMultipartPost-642b1a8c3f4afb7658ab808cc646bc9e\r\n" +
App 5065 output:      "Content-Disposition: form-data; name=\"attachment[]\"; filename=\"Ticket\"\r\n" +
App 5065 output:      "Content-Length: 37142\r\n" +
App 5065 output:      "Content-Type: application/pdf\r\n" +
App 5065 output:      "Content-Transfer-Encoding: binary\r\n" +
App 5065 output:      "\r\n",
App 5065 output:     @io=
App 5065 output:      #<CompositeReadIO:0x00007f96d9fd8a78
App 5065 output:       @index=0,
App 5065 output:       @ios=
App 5065 output:        [#<StringIO:0x00007f96d9fd8ac8>,
App 5065 output:         #<UploadIO:0x00007f96d9f32f60
App 5065 output:          @content_type="application/pdf",
App 5065 output:          @io=
App 5065 output:           #<File:/vagrant_ruby/martian-library/pdfs/Ticket-128DBBAE-60D4-8CF2-0D8B-7D9E3A9B912D.pdf>,
App 5065 output:          @local_path=
App 5065 output:           "/vagrant_ruby/martian-library/pdfs/Ticket-128DBBAE-60D4-8CF2-0D8B-7D9E3A9B912D.pdf",
App 5065 output:          @opts={},
App 5065 output:          @original_filename="Ticket">,
App 5065 output:         #<StringIO:0x00007f96d9fd8aa0>]>,
App 5065 output:     @length=37372>,
App 5065 output:    #<Parts::FilePart:0x00007f96d9fd8938
App 5065 output:     @foot="\r\n",
App 5065 output:     @head=
App 5065 output:      "-------------RubyMultipartPost-642b1a8c3f4afb7658ab808cc646bc9e\r\n" +
App 5065 output:      "Content-Disposition: form-data; name=\"attachment[]\"; filename=\"Voucher\"\r\n" +
App 5065 output:      "Content-Length: 33454\r\n" +
App 5065 output:      "Content-Type: application/pdf\r\n" +
App 5065 output:      "Content-Transfer-Encoding: binary\r\n" +
App 5065 output:      "\r\n",
App 5065 output:     @io=
App 5065 output:      #<CompositeReadIO:0x00007f96d9fd8640
App 5065 output:       @index=0,
App 5065 output:       @ios=
App 5065 output:        [#<StringIO:0x00007f96d9fd8690>,
App 5065 output:         #<UploadIO:0x00007f96d9fdb750
App 5065 output:          @content_type="application/pdf",
App 5065 output:          @io=
App 5065 output:           #<File:/vagrant_ruby/martian-library/pdfs/Voucher-128DBBAE-60D4-8CF2-0D8B-7D9E3A9B912D.pdf>,
App 5065 output:          @local_path=
App 5065 output:           "/vagrant_ruby/martian-library/pdfs/Voucher-128DBBAE-60D4-8CF2-0D8B-7D9E3A9B912D.pdf",
App 5065 output:          @opts={},
App 5065 output:          @original_filename="Voucher">,
App 5065 output:         #<StringIO:0x00007f96d9fd8668>]>,
App 5065 output:     @length=33685>,
App 5065 output:    #<Parts::ParamPart:0x00007f96d9fd8528
App 5065 output:     @io=#<StringIO:0x00007f96d9fd83e8>,
App 5065 output:     @part=
App 5065 output:      "-------------RubyMultipartPost-642b1a8c3f4afb7658ab808cc646bc9e\r\n" +
App 5065 output:      "Content-Disposition: form-data; name=\"data\"\r\n" +
App 5065 output:      "Content-Type: application/json\r\n" +
App 5065 output:      "\r\n" +
App 5065 output:      "{\"template\":\"reports_email\",\"emailTo\":\"adrian.meza@netpay.com.mx\",\"emailData\":{\"store_name\":\"MAKER\",\"type\":\"Ticket\"}}\r\n">,
App 5065 output:    #<Parts::EpiloguePart:0x00007f96d9fd83c0
App 5065 output:     @io=#<StringIO:0x00007f96d9fd8370>,
App 5065 output:     @part=
App 5065 output:      "-------------RubyMultipartPost-642b1a8c3f4afb7658ab808cc646bc9e--\r\n">]>
App 5065 output:
App 5065 output: I, [2020-07-21T16:33:59.953913 #5065]  INFO -- response: Status 200
App 5065 output: I, [2020-07-21T16:33:59.954231 #5065]  INFO -- response: content-length: "0"
App 5065 output: date: "Tue, 21 Jul 2020 16:33:59 GMT"

**Comment by iMacTia** [comment]
Thank you @adrian-netapay. As you can see, the files are all there.
I suspect your issue might be related to the parameter name becoming `attachment[]` (note the `[]`) when you send more than one file, where the server you call might be expecting an old-style `attachment` instead.

Good news is that this issue was recently raised (https://github.com/lostisland/faraday/issues/1157) and a PR to support this was also merged (https://github.com/lostisland/faraday/pull/1163).

Try adding the `flat_encode` options to the `multipart` middleware in your connection configuration:

```ruby
conn = Faraday.new() do |f|
  f.request :multipart, flat_encode: true
  f.adapter :net_http
end
```

**Comment by iMacTia** [comment]
@adrian-netapay oh, I almost forgot, this PR was merged very recently, so it hasn't been released yet and it won't work on Faraday 1.0.1

Can you test this out using Faraday's master branch?

**Comment by iMacTia** [comment]
Glad to hear that 🎉 !
I'm closing this one then 👍
